internal/config/config.go

```diff
@@ -50,6 +50,7 @@ type SubmitConfig struct {
 	Address    string `yaml:"address"      envconfig:"SUBMIT_TCP_ADDRESS"`
 	SocketPath string `yaml:"socketPath"   envconfig:"SUBMIT_SOCKET_PATH"`
 	Url        string `yaml:"url"          envconfig:"SUBMIT_URL"`
+	UrlKey     string `yaml:"urlKey"       envconfig:"SUBMIT_URL_KEY"`
```

Comment:
This is ambiguous. Key for who? What? This would be better handled with generic "header" support, rather than a provider specific implementation.